llvm/lib/CodeGen/ImplicitNullChecks.cpp

```diff
@@ -346,6 +346,8 @@ ImplicitNullChecks::areMemoryOpsAliased(const MachineInstr &MI,
           return AR_MayAlias;
         continue;
       }
+      if (MMO2->getValue() == nullptr)
+        continue;
```

Comment:
Spent a bit of time reading more closely. This case is hit if of the previous instructions has a memory operand that has neither a `PseudoSourceValue` nor a `Value`. Maybe it'd be safest, for now, to assume that PrevMI and MI operate on aliased memory (e.g. just `return AR_MayAlias`)